### PR TITLE
Add test for Ecotone

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Add a test to ensure that Ecotone occurs on a block boundary.

### CODE REVIEW

**Review by nettucloud — APPROVED** [review]

File: superchain/superchain_test.go [hunk 1]
```diff
@@ -286,18 +287,20 @@ func TestContractBytecodes(t *testing.T) {
...
+			chainCfg := OPChains[id]
+			t.Run(fmt.Sprintf("%s_%s", superchainName, chainCfg.Name), testNetworkUpgradeTimestampOffset(chainCfg.Genesis.L2Time, 2, superchainConfig.Config.CanyonTime))
 		}
-		ct := *superchainConfig.Config.CanyonTime
+	}
+}
+
+// TestEcotoneTimestampOnBlockBoundary asserts that Ecotone will activate on a block's timestamp.
+// This is critical because the L2 upgrade transactions only activates on a block's timestamp.
+func TestEcotoneTimestampOnBlockBoundary(t *testing.T) {
```
Comment (sebastianst) [inline superchain/superchain_test.go:299]: We could create a generic test function for this to avoid duplication in the future for standard chains, like
```go
func testStandardTimestampOnBlockBoundary(t *testing.T, desc string, ts func(Superchain) *uint64 ) {
	for superchainName, superchainConfig := range Superchains {
		for _, id := range superchainConfig.ChainIDs {
			chainCfg := OPChains[id]
			t.Run(fmt.Sprintf("%s/%s/%s", superchainName, chainCfg.Name, desc), testNetworkUpgradeTimestampOffset(chainCfg.Genesis.L2Time, 2, ts(superchainConfig))
		}
	}
}
```
and then call it for `Canyon` and `Ecotone`.

**Review by sebastianst — APPROVED** [review]
LGTM just one nit and one optional suggestion.

**Review by protolambda — APPROVED** [review]
LGTM after implementing suggestions of Seb

**Review by sebastianst — APPROVED** [review]
